plugins/postgres/dbt/adapters/postgres/relation_configs/materialized_view.py

```diff
@@ -0,0 +1,129 @@
+                ),
+            ),
+        }
+
+    @classmethod
+    def from_dict(cls, config_dict: dict) -> "PostgresMaterializedViewConfig":
+        kwargs_dict = {
+            "table_name": config_dict.get("table_name"),
+            "query": config_dict.get("query"),
+            "indexes": {
+                PostgresIndexConfig.from_dict(index) for index in config_dict.get("indexes", {})
+            },
+        }
+        materialized_view: "PostgresMaterializedViewConfig" = super().from_dict(kwargs_dict)  # type: ignore
+        return materialized_view
+
+    @classmethod
+    def from_model_node(cls, model_node: ModelNode) -> "PostgresMaterializedViewConfig":
+        materialized_view_config = cls.parse_model_node(model_node)
+        materialized_view = cls.from_dict(materialized_view_config)
+        return materialized_view
+
+    @classmethod
+    def parse_model_node(cls, model_node: ModelNode) -> dict:
+        """
+        Postgres-specific implementation of `RelationConfig.from_model_node()` for materialized views
+        """
+        config_dict = {
+            "table_name": model_node.identifier,
+            "query": model_node.compiled_code,
+        }
+
+        # create index objects for each index found in the config
+        if indexes := model_node.config.extra.get("indexes"):
+            index_configs = [PostgresIndexConfig.parse_model_node(index) for index in indexes]
+            config_dict.update({"indexes": index_configs})
+
+        return config_dict
+
+    @classmethod
+    def from_relation_results(
+        cls, relation_results: RelationResults
+    ) -> "PostgresMaterializedViewConfig":
+        materialized_view_config = cls.parse_relation_results(relation_results)
+        materialized_view = cls.from_dict(materialized_view_config)
+        return materialized_view
+
+    @classmethod
+    def parse_relation_results(cls, relation_results: RelationResults) -> dict:
+        """
+        Postgres-specific implementation of `RelationConfig.from_relation_results()` for materialized views
+        """
+        base_config = relation_results.get("base", {})
+        config_dict = {
+            "table_name": base_config.get("table_name"),
+            "query": base_config.get("query"),
+        }
+
+        # create index objects for each index found in the config
+        if indexes := relation_results.get("indexes"):
+            index_configs = [
+                PostgresIndexConfig.parse_relation_results({"base": index})
+                for index in indexes.rows
+            ]
+            config_dict.update({"indexes": index_configs})
+
+        return config_dict
+
+
+@dataclass
+class PostgresMaterializedViewConfigChangeCollection:
+    indexes: Optional[Set[PostgresIndexConfigChange]] = None
+
+    @property
+    def requires_full_refresh(self) -> bool:
+        return any(index.requires_full_refresh for index in self.indexes)
```

Comment:
this probably needs the same wrapping of `if isinstance(self.indexes, Iterable)` that `has_changes` has because `indexes` is optional

Reply:
Done